tests/test_datasets/test_dataset_functions.py

```diff
@@ -1316,3 +1316,33 @@ def test_list_qualities(self):
         qualities = openml.datasets.list_qualities()
         self.assertEqual(isinstance(qualities, list), True)
         self.assertEqual(all([isinstance(q, str) for q in qualities]), True)
+
+    def test_get_dataset_cache_format_pickle(self):
+        # Feather format cant be tested without installing pyarrow
+        # this test case checks if pickle option works
+        dataset = openml.datasets.get_dataset(1)
+        self.assertEqual(type(dataset), OpenMLDataset)
+        self.assertEqual(dataset.name, 'anneal')
+        self.assertGreater(len(dataset.features), 1)
+        self.assertGreater(len(dataset.qualities), 4)
+
+        X, y, categorical, attribute_names = dataset.get_data()
+        self.assertIsInstance(X, pd.DataFrame)
+        self.assertEqual(X.shape, (898, 39))
+        self.assertEqual(len(categorical), X.shape[1])
+        self.assertEqual(len(attribute_names), X.shape[1])
+
+    def test_get_dataset_cache_format_feather(self):
+        # Feather format cant be tested without installing pyarrow
+        # this test case checks if pickle option works
+        dataset = openml.datasets.get_dataset('iris', cache_format='feather')
```

Comment:
Could you please add a test that the correct files are actually written to disk?

Reply:
I'm afraid this doesn't test the right thing. We change the cache directory for unit tests, so you'd need to use a slightly different data_folder here (https://github.com/openml/openml-python/blob/develop/openml/testing.py#L92)